clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -197,30 +218,58 @@ record_filter_t::parallel_shard_memref(void *shard_data, const trace_entry_t &in
             // XXX: it may happen that we never output a unit header due to this
             // optimization. We should ensure that we output it at least once. We
             // skip handling this corner case for now.
-            per_shard->last_filtered_unit_header.clear();
+            per_shard->last_delayed_unit_header.clear();
             ANNOTATE_FALLTHROUGH;
         case TRACE_MARKER_TYPE_WINDOW_ID:
         case TRACE_MARKER_TYPE_CPU_ID:
             if (output)
-                per_shard->last_filtered_unit_header.push_back(entry);
-            output = false;
+                per_shard->last_delayed_unit_header.push_back(entry);
+            return true;
         }
     }
+
+    if (!output) {
+        if (is_any_instr_type(static_cast<trace_type_t>(entry.type)) &&
+            !per_shard->last_delayed_encoding.empty()) {
+            // Overwrite in case the encoding for this pc was already recorded.
+            per_shard->delayed_encodings[entry.addr] =
+                std::move(per_shard->last_delayed_encoding);
+            per_shard->last_delayed_encoding = {};
```

Comment:
Since this was `std::move`d above, we shouldn't be calling operations on the vector object. Assigning a new one is best I think.

Reply:
I believe you can still rely on clear() working.  Some evidence: https://stackoverflow.com/questions/49723585/is-it-legal-to-call-clear-shrink-to-fit-empty-on-a-moved-from-stdve

Reply:
I also found this link that confirms that clear() is considered a valid re-initialization for standard containers: https://releases.llvm.org/11.1.0/tools/clang/tools/extra/docs/clang-tidy/checks/bugprone-use-after-move.html#unsequenced-moves-uses-and-reinitializations.

> .clear() is probably faster?

This is probably true in general because the vector doesn't need to reallocate memory. But not sure about the case after an `std::move`, where the vector's resources (the allocated mem) has already been taken away from it.